script.js

```diff
@@ -174,61 +175,27 @@ function checkBadFood(arr, direction, snakeBlocks) {
   }
 }
 
-function moveRight(snakeBlocks) {
-  if (touchingEdge("right", snakeBlocks) || touchingSelf()) {
-    die();
-  }
-  var foreSquare = snakeBlocks[snakeBlocks.length - 1];
-
-  snakeBlocks = checkFood(food, "right", snakeBlocks);
-  checkBadFood(badFood, "right", snakeBlocks);
-  snakeBlocks.push([foreSquare[0] + 1, foreSquare[1]]);
-  return snakeBlocks;
-}
-
-function moveLeft(snakeBlocks) {
-  if (touchingEdge("left", snakeBlocks) || touchingSelf()) {
-    die();
-  }
-  snakeBlocks = checkFood(food, "left", snakeBlocks);
-  checkBadFood(badFood, "left", snakeBlocks);
-  var foreSquare = snakeBlocks[snakeBlocks.length - 1];
-  snakeBlocks.push([foreSquare[0] - 1, foreSquare[1]]);
-  return snakeBlocks;
-}
-
-function moveUp(snakeBlocks) {
-  if (touchingEdge("up", snakeBlocks) || touchingSelf()) {
-    die();
-  }
-  snakeBlocks = checkFood(food, "up", snakeBlocks);
-  checkBadFood(badFood, "up", snakeBlocks);
-  var foreSquare = snakeBlocks[snakeBlocks.length - 1];
-  snakeBlocks.push([foreSquare[0], foreSquare[1] - 1]);
-  return snakeBlocks;
-}
-
-function moveDown(snakeBlocks) {
-  if (touchingEdge("down", snakeBlocks) || touchingSelf()) {
+function move(snakeBlocks, direction, upDown, rightLeft){
+  if (touchingEdge(direction, snakeBlocks) || touchingSelf()) {
```

Comment:
Note, that the way you're using `touchingEdge` and `touchingSelf` here they should return a boolean. See my comments at those functions.